doc/FAQ.md

```diff
@@ -0,0 +1,102 @@
+
+If there are more than one tasks but only a single machine is working make sure that the job itself
+has been allocated multiple executor slots to work with. This is set at the time of SparkContext
+creation with `spark.cores.max` in the `SparkConf` and cannot be changed during the job. 
+
+### Why can't the spark job find Spark Cassandra Connector Classes? (ClassNotFound Exceptions for SCC Classes)
+
+The most common cause for this is that the executor classpath does not contain the Spark Cassandra Connector
+jars. The simplest way to add these to the class path is to use SparkSubmit with the --jars option pointing
+to your Spark Cassandra Connector assembly jar. If this is impossible for some reason the second best option
+is to manually distribute the jar to all of your executors and add the jar's location to the executor classPath
+in the SparkConf or spark-defaults.conf.
+
+### Where should I set configuration options for the connector?
+
+The suggested location is to use the `spark-defaults.conf` file in your spark/conf directory but 
+this file is ONLY used by spark-submit. Any applications not running through spark submit will ignore
+this file. 
+ 
+For those you should set the options in the `SparkConf` object used to 
+create your `SparkContext`. Usually this will take the form of a series of statements that look like
+
+    val conf = SparkConf()
+      .set("Option","Value")
+      ...
+      
+    val sc = SparkContext(conf)
+
+### Why are my write tasks timing out/ failing?
+
+The most common cause of this is that Spark is able to issue write requests much more quickly than
+Cassandra can handle them. This can lead to GC issues and build up of hints. If this is the case
+with your application try lowering the number of concurrent writes and the current batch size using
+the following options.
+
+   spark.cassandra.output.batch.size.rows
+   spark.cassandra.output.concurrent.writes
+   
+### Why are my executors throwing `OutOfMemoryException`s while Reading from Cassandra?
+
+This is usually means that the size of the partitions you are attempting to create are larger than
+the executor's heap can handle. Remember that all of the executors run in the same JVM so the size
+of the data is multiplied by the number of executor slots.
+
+To fix this either increase the heap size of the executors `spark.executor.memory`
+ or shrink the size of the partitions by decreasing `spark.cassandra.input.split.size`
+ 
+### Why can't my spark job find My Application Classes / Anonymous Functions?
+ 
+This occurs when your application code hasn't been placed on the classpath of the Spark Executor. When using
+Spark Submit make sure that the jar contains all of the classes and dependencies for running your code. 
+To build a fat jar look into using sbt assembly, or look for instructions for your build tool of choice.
+
+If you are not using Spark Submit (not recommended) make sure that your dependencies have been set
```

Comment:
The 'not recommended' part could be confusing here. Suggest reword like "If you are not using the recommended approach with Spark Submit, make sure...".